lib/app_component.dart

```diff
@@ -380,7 +379,7 @@ class AppComponent {
       equationItems.add(totalValue);
 
       //this fixes precision issues sqrt(2)*sqrt(2) = 2.000000000004
-      if (totalValue.toString().contains(".")) {
+      if (totalValue.toString().contains(decimal)) {
```

Comment:
I want to take it a step deeper and get rid of the fact that you have to truncate at all. I know there was a 3rd party BigDecimal-esque class in Dart. Why couldn't you use that?